src/Sulu/Bundle/DocumentManagerBundle/DependencyInjection/Compiler/SubscriberPass.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\DocumentManagerBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Compiler pass to register document manager subscribers.
+ *
+ * Subscriber tags may indicate to which document manager they
+ * should be associated.
+ */
+class SubscriberPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $managers = $container->getParameter('sulu_document_manager.managers');
+        $baseDispatcherId = 'sulu_document_manager.event_dispatcher';
+        $dispatcherIds = [];
+
+        foreach ($managers as $inde => $manager) {
+            $dispatcherId = $baseDispatcherId . '.' . $manager;
+
+            // TODO: Why would this happen?
+            if (!$container->hasDefinition($dispatcherId) && !$container->hasAlias($dispatcherId)) {
+                continue;
```

Comment:
The extension should have already created the dispatcher, we sohuld throw an exception in this case.